tests/form.js

```diff
@@ -36,6 +36,56 @@ define([
             });
         });
 
+        it("should get and fix serialization errors for mugs with matching paths", function () {
+            var form = util.loadXML(""),
+                one = util.addQuestion("Text", "question"),
+                two = util.addQuestion("Text", "question");
+            assert.notEqual(one.absolutePath, two.absolutePath);
+            var errors = form.getSerializationWarnings();
+            assert.equal(errors.length, 1, "missing serialization error message");
+            assert.equal(errors[0].mug.ufid, two.ufid);
+
+            form.fixSerializationWarnings(errors);
+            assert.equal(one.p.nodeID, "question");
+            assert.notEqual(one.absolutePath, two.absolutePath);
+            assert.deepEqual(form.getSerializationWarnings(), []);
+        });
+
+        it("should retain expression meaning on rename matching path", function () {
+            var blue = util.addQuestion("Text", "blue"),
+                green = util.addQuestion("Text", "green"),
+                black = util.addQuestion("DataBindOnly", "black");
+            black.p.calculateAttr = "/data/blue + /data/green";
+            green.p.nodeID = "blue";
+            assert.notEqual(green.p.nodeID, "blue");
+            assert(!util.isTreeNodeValid(green), "expected validation error");
+
+            blue.p.nodeID = "orange";
+            assert.equal(green.p.nodeID, "blue");
+            assert.equal(black.p.calculateAttr, "/data/orange + /data/blue");
```

Comment:
Had to go back to my original plan of saving the conflicting ID on the mug rather than allowing duplicate IDs in the tree. Otherwise these two paths would both end up pointing to `/data/orange`.